plugins/builds/update.js

```diff
@@ -53,20 +54,23 @@ module.exports = () => ({
                         }
                         // Check permission against the pipeline
                         // @TODO implement this
-                    } else {
-                        switch (desiredStatus) {
-                        case 'SUCCESS':
-                        case 'FAILURE':
-                        case 'ABORTED':
-                            build.meta = request.payload.meta || {};
-                            build.endTime = (new Date()).toISOString();
-                            break;
-                        case 'RUNNING':
-                            build.startTime = (new Date()).toISOString();
-                            break;
-                        default:
-                            throw boom.badRequest(`Cannot update builds to ${desiredStatus}`);
-                        }
+                    }
+
+                    return eventFactory.get(build.eventId).then(event => ({ build, event }));
```

Comment:
I think `.then(event => ({ build, event }))` can be skipped, but it's more readable if we keep it the way you have it.